docs/api/Scan.md

```diff
@@ -1,6 +1,8 @@
 # Scan
 
-Dynamoose provides the ability to scan a model by using the `Model.scan` function. This function acts as a builder to construct your scan with the appropriate settings before executing it (`scan.exec`).
+Dynamoose provides the ability to scan a table (model) by using the `Model.scan` function. This function acts as a builder to construct your scan with the appropriate settings before executing it (`scan.exec`).
+
+**Please note:** The Scan operation operates on your entire table. For tables of real size, this can quickly use up all of your Read Capacity. If you're using it in your application's critical path, it will be very slow in returning a response to your users. The best option is not never use `scan()` unless you know what you are doing! 
```

Comment:
> **Please note:**

Not a fan of the bold for same reasons as I discussed above.

> The Scan operation operates on your entire table.

This is not always true. It is possible to run a Scan on a secondary index, which has the potential to have a different number of items than the table itself.

I understand what you mean here, but I think there is a layer of confusion this could create.

> If you're using it in your application's critical path, it will be very slow in returning a response to your users.

I don't think this is always true either. Unless you are referring to large tables from above. But a table with let's say 3 items, that you always run a scan on, that very likely won't be very slow in returning a response to your users.

I do not believe Dynamoose should be giving database design or modeling advice. Every single use case is different. I believe we should present the information, but not make recommendations here. That opens a whole other layer of support if there is a use case for it, and I can totally imagine the emails and questions about `You said this was a bad idea, but I have this legit use case. Why is it a bad idea?`.

So I'm really against making assumptions about use cases or applications for Dynamoose.

> The best option is not never use `scan()` unless you know what you are doing!

Same reason here. Saying something along the lines of `The Scan operation operates on your entire table.` I believe makes this clear. I'm not a fan of being opinionated here. There are many very legit use cases for Scan. It's important to know what you are doing for sure. But isn't that true for many applications? I think the important part is that it has no partitions or filtering before looking through the items. It scans everything. I think that gets the point across.

I think we should leave the education about what Scan does behind the scenes and discouraging it to AWS. The goal of this documentation is to explain Dynamoose, not DynamoDB.

I think the best example of this, is you don't see [Express.js](https://expressjs.com) explaining how the web works. I mean even in their [`res.status`](https://expressjs.com/en/4x/api.html#res.status) documentation they don't even explain that 2xx means success, 4xx means client error, or 5xx means server error. They give the information about how Express works and leaves the rest to you to research on your own.

Reply:
I copied this 1 to 1 from https://www.dynamodbguide.com/scans , written by a DynamoDB expert. IMO in a case where the user can do very bad things we should give an advice. I mean the scan function is there, everyone can use it, but he has been warned at least...